src/components/cdTab.js

```diff
@@ -1,35 +1,187 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 
-import Grid from '@mui/material/Grid'
-import { Skeleton } from '@mui/material'
+import {
+  Alert,
+  Box,
+  FormControlLabel,
+  FormGroup,
+  Switch,
+  Tooltip,
+} from '@mui/material'
 
 import MultilineTextViewer from './util/MultilineTextViewer'
+import { useFetchComponentDescriptor } from '../api/useFetch'
+import CenteredSpinner from './util/CenteredSpinner'
+import { SearchParamContext } from '../App'
 import { toYamlString } from '../util'
 
 
+const FetchAndView = ({
+  componentName,
+  componentVersion,
+  ocmRepo,
+  versionFilter,
+  setRaw,
+}) => {
+  // eslint-disable-next-line no-unused-vars
+  const [componentDescriptor, isLoading, isError, _] = useFetchComponentDescriptor({
+    componentName: componentName,
+    ocmRepoUrl: ocmRepo,
+    version: componentVersion,
+    versionFilter: versionFilter,
+    raw: true,
+  })
+
+  React.useEffect(() => {
+    setRaw(componentDescriptor)
+  }, [setRaw, componentDescriptor])
+
+  return <Viewer
+    isError={isError}
+    componentDescriptor={JSON.stringify(componentDescriptor, null, 2)}
+  />
+}
+FetchAndView.displayName = 'FetchAndView'
+FetchAndView.propTypes = {
+  componentName: PropTypes.string.isRequired,
+  componentVersion: PropTypes.string.isRequired,
+  ocmRepo: PropTypes.string,
+  versionFilter: PropTypes.string,
+  setRaw: PropTypes.func.isRequired,
+}
+
+
+const Options = ({
+  isLoading,
+  showRaw,
+  setShowRaw,
+}) => {
+  return <Box>
+    <FormGroup>
+      <FormControlLabel
+        control={
+          <Tooltip
+            title='Raw document as retrieved from OCM repository, without (de)serialisation'
+          >
+            <Switch
+              checked={showRaw}
+              // eslint-disable-next-line no-unused-vars
+              onChange={(event) => setShowRaw((prev) => !prev)}
+              disabled={isLoading}
+            />
+          </Tooltip>
+        }
+        label='show raw'
+      />
+    </FormGroup>
+  </Box>
+}
+Options.displayName = 'Options'
+Options.propTypes = {
+  isLoading: PropTypes.bool,
+  showRaw: PropTypes.bool.isRequired,
+  setShowRaw: PropTypes.func.isRequired,
+}
+
+
+const Viewer = ({
+  componentDescriptor,
+  isError
+}) => {
+  if (isError) return <Alert severity='critical'>
+    Something went wrong
+  </Alert>
+
+  return <MultilineTextViewer
+    text={componentDescriptor}
+  />
+}
+Viewer.displayName = 'Viewer'
+Viewer.propTypes = {
+  componentDescriptor: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
```

Comment:
`componentDescriptor` should only be of type `string` now with the recent refactoring to the `MultiLineTextViewer`